### PR TITLE
AP_HAL_ChibiOS: IOMCU digital out

### PR DESCRIPTION
**Pull request by IamPete1**
This work is supported by CubePilot.

This adds support outputs via the IOMCU. For relay ect.

We could theoretically also do inputs, but there is a one way buffer preventing it in hardware.

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/GPIO.cpp [hunk 1]
```diff
@@ -73,6 +79,13 @@ void GPIO::init()
     uint8_t chan_offset = 0;
 #if HAL_WITH_IO_MCU
     if (AP_BoardConfig::io_enabled()) {
+        uint8_t GPIO_mask = 0;
+        for (uint8_t i=0; i<8; i++) {
+            if (SRV_Channels::is_GPIO(i)) {
+                GPIO_mask |= 1 << i;
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/GPIO.cpp:85]: I prefer 1U << i so it is clearly unsigned. Doesn't matter in this case, but good habit

**Review by tridge — CHANGES_REQUESTED** [review]
really nice, thanks! just minor issues
did you use 10.2 compiler to build IO fw?

**Comment by IamPete1** [comment]
Io-firmware built with `g++ 10.2.1`. just re-tested on CubeOrange

**Review by tridge — APPROVED** [review]

**Comment by IamPete1** [comment]
Costs `184` bytes on the IOfirmware, now up to `47984`

**Comment by tridge** [comment]
check memory usage, is it still allocation all the RC backends?

**Comment by IamPete1** [comment]
As far as I can tell this uses no more memory than before. `IOMC.Mem` is 704 on both master and this.

**Review by tridge — APPROVED** [review]

**Pull request merged**